src/Interpreters/ClusterProxy/SelectStreamFactory.cpp

```diff
@@ -173,5 +172,106 @@ void SelectStreamFactory::createForShard(
         emplace_remote_stream();
 }
 
+
+SelectStreamFactory::ShardPlans SelectStreamFactory::createForShardWithParallelReplicas(
+    const Cluster::ShardInfo & shard_info,
+    const ASTPtr & query_ast,
+    const StorageID & main_table,
+    const ASTPtr & table_function_ptr,
+    const ThrottlerPtr & throttler,
+    ContextPtr context,
+    UInt32 shard_count)
+{
+    SelectStreamFactory::ShardPlans result;
+
+    auto it = objects_by_shard.find(shard_info.shard_num);
+    if (it != objects_by_shard.end())
+        replaceMissedSubcolumnsByConstants(storage_snapshot->object_columns, it->second, query_ast);
+
+    const auto & settings = context->getSettingsRef();
+
+    auto is_local_replica_obsolete = [&]()
+    {
+        auto resolved_id = context->resolveStorageID(main_table);
+        auto main_table_storage = DatabaseCatalog::instance().tryGetTable(resolved_id, context);
+
+        if (!main_table_storage)
+            throw DB::Exception(ErrorCodes::ALL_REPLICAS_ARE_STALE, "");
```

Comment:
Do not understand.
In which situation can we get nullptr storage from catalog? Why it means that all replicas are stale?
Exception message is empty.